bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BatchExecutor.java

```diff
@@ -72,77 +71,54 @@
         }
       };
 
-  private static final Function<GeneratedMessage, Object> ROW_RESULT_CONVERTER =
-      new Function<GeneratedMessage, Object>() {
-
-        @Override
-        public Object apply(GeneratedMessage response) {
-          if (response instanceof com.google.bigtable.v1.Row) {
-            return Adapters.ROW_ADAPTER.adaptResponse((com.google.bigtable.v1.Row) response);
-          } else {
-            return new Result();
-          }
-        }
-      };
-
   /**
    * A callback for ListenableFutures issued as a result of an RPC
-   * @param <R>
-   * @param <T> The response messsage type.
    */
-  static class RpcResultFutureCallback<R, T extends GeneratedMessage>
-      implements FutureCallback<T> {
-
+  static class RpcResultFutureCallback<T> implements FutureCallback<GeneratedMessage> {
     private final Row row;
-    private final Batch.Callback<R> callback;
+    private final Batch.Callback<T> callback;
     private final int index;
     private final Object[] resultsArray;
-    private final SettableFuture<Object> resultFuture;
-    private final Function<T, Object> adapter;
+    private final SettableFuture<Result> resultFuture;
```

Comment:
can we just use SettableFuture instead of both resultsArray + index? Once resultFuture.get() return the result, they don't need to look at resultsArray, no?